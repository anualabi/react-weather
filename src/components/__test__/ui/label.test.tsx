import { render, screen } from "@testing-library/react";

import Label, { LabelProps } from "@/components/ui/label";

const mockProps = {
  children: "Test Label",
  className: "",
};

const customRender = (props: LabelProps) => {
  return render(<Label {...props} />);
};

describe("Label", () => {
  it("should render (snapshot)", () => {
    const { asFragment } = customRender({ ...mockProps });
    expect(asFragment()).toMatchSnapshot();
  });

  it("shold render a label HTML element", () => {
    customRender({ ...mockProps });
    const labelElement = screen.getByText("Test Label");
    expect(labelElement.tagName).toBe("LABEL");
  });

  it("should render its children", () => {
    customRender({ ...mockProps });
    const labelElement = screen.getByText(mockProps.children);
    expect(labelElement).toBeInTheDocument();
  });
});
