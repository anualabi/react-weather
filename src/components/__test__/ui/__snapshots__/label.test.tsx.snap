// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render (snapshot) 1`] = `
<DocumentFragment>
  <label
    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
  >
    Test Label
  </label>
</DocumentFragment>
`;
